### PR TITLE
MAINT: Update master to main after branch rename

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
                   --message "Docs build of $CIRCLE_SHA1" \
                   --force
             else
-              echo "Not on the master branch; skipping deployment"
+              echo "Not on the main branch; skipping deployment"
```
Comment (rossbar) [inline .circleci/config.yml:103]: Do the references in the opening conditional also need to be updated: `"master"` -> `"main"` too?

Comment (stefanv) [reply .circleci/config.yml:104]: Good catch, fixed.

**Review by seberg — APPROVED** [review]
LGTM, thanks Stéfan, I suspect this is breaking some things, so happy if you or Ross just merge it when content.

**Review by rossbar — APPROVED** [review]
LGTM too, looks like this gets most of the instances. I'd like to double-check the CI logs as well when they complete to see if there's anything obvious

**Comment by mattip** [comment]
Azure is stuck with "This agent request is not running because you have reached the maximum number of requests that can run for parallelism type", restarting it did not help.

**Comment by rgommers** [comment]
> Azure is stuck with "This agent request is not running because you have reached the maximum number of requests that can run for parallelism type", restarting it did not help.

There's a massive backlog, but I don't see how to cancel all running jobs at once.

**Review by rgommers — APPROVED** [review]
LGTM too, thanks @stefanv!

**Comment by seberg** [comment]
Going to merge this, before it creates some problems.  That hanging test is weird, but not really worrisome. Thanks again!

**Comment by charris** [comment]
The azure pipeline is also trying to read master. I edited the pipeline yml, lets see if that helps. If not, the easiest fix may be to delete the old pipeline and make a new one.

**Comment by charris** [comment]
Wheel builds will need fixing, they currently keep a submodule with the numpy master branch.

**Comment by seberg** [comment]
We should probably rename to main in numpy-wheels as well. It seems much cleaner if the branch it builds has the identical name (I do not have access to rename) – otherwise have to carefully distinguish whether the occurance of "master" refers to the wheel repo or the numpy one, making the change more than a trivial replace.

**Comment by charris** [comment]
@seberg I'm going to fix the wheel build in two steps

- Fix the yml files
- Do the rename

The first step is running now, we will see how it goes. The NumPy submodule should be OK, it points to a hash. I checked that I could also checkout main from numpy after doing `git submodule sync --recursive`. I'm going to think a bit as to how to do the branch rename in order to avoid the current Numpy problems.

**Comment by charris** [comment]
Turns out that the job limit applies to all the pipelines, so the azure wheels build is also stuck waiting.

**Comment by seberg** [comment]
Microsoft isn't letting me log into azure easily (claiming my private email is a job email). It seems it restarted the jobs for all PRs on azure, not sure if we can clear the whole pipeline somehow?  I guess on wheel repo that shouldn't trigger hundrets of jobs at least...

**Comment by charris** [comment]
Some jobs are running, so maybe we just need to wait.

**Comment by charris** [comment]
The US has had several days of degraded azure performance: https://status.dev.azure.com/_history. Seems to be a azire problem.

**Comment by stefanv** [comment]
Thanks for helping with the loose threads @seberg @charris @rossbar

**Comment by charris** [comment]
I'm going to suggest that if this is done for SciPy that it might be a good idea to disable running tests on PRs before making the switch, otherwise azure will probably spend a couple of days rerunning tests on all the outstanding PRs.

**Comment by charris** [comment]
Or maybe set up a second pipeline and point it at "main". I'm not quite sure what we did to get all the retests.

**Pull request merged**